app/views/mod.py

```diff
@@ -0,0 +1,225 @@
+""" Mod endpoints """
+import time
+import re
+from peewee import fn, JOIN, Value
+from pyotp import TOTP
+from flask import Blueprint, abort, redirect, url_for, session, render_template, jsonify
+from flask_login import login_required, current_user
+from flask_babel import _
+from .. import misc
+from ..forms import TOTPForm, LogOutForm, UseInviteCodeForm, AssignUserBadgeForm, EditModForm, BanDomainForm
+from ..models import UserMetadata, User, Sub, SubPost, SubPostComment, SiteMetadata
+from ..models import User, Sub, SubMod, SubPost, SubPostComment, UserMetadata, SubPostReport, SubPostCommentReport
+from ..misc import engine, getSubReports, getModSubs, engine
+
+bp = Blueprint('mod', __name__)
+
+
+@bp.route('/mod/auth', methods=['GET', 'POST'])
+@login_required
+def auth():
+    if not current_user.is_mod:
+        abort(404)
+    form = TOTPForm()
+    try:
+        user_secret = UserMetadata.get((UserMetadata.uid == current_user.uid) & (UserMetadata.key == 'totp_secret'))
+    except UserMetadata.DoesNotExist:
+        return engine.get_template('mod/totp.html').render({'authform': form, 'error': _('No TOTP secret found.')})
+    if form.validate_on_submit():
+        totp = TOTP(user_secret.value)
+        if totp.verify(form.totp.data):
+            session['apriv'] = time.time()
+            return redirect(url_for('mod.index'))
+        else:
+            return engine.get_template('mod/totp.html').render(
+                {'authform': form, 'error': _('Invalid or expired token.')})
+    return engine.get_template('mod/totp.html').render({'authform': form, 'error': None})
+
+
+@bp.route('/logout', methods=['POST'])
+@login_required
+def logout():
+    if not current_user.is_mod:
+        abort(404)
+    form = LogOutForm()
+    if form.validate():
+        del session['apriv']
+    return redirect(url_for('mod.index'))
+
+
+@bp.route("/")
+@login_required
+def index():
+    """ WIP: Mod Dashboard """
+
+    if not current_user.is_mod:
```

Comment:
This seems to be only checking if the `is_mod` function exists... it should be removed anyway since anybody should be able to enter here

Reply:
Good catch. I do think it's better if regular users can't even find this view by accident, though, so updated this to:

```
    if not SubMod.select().where(SubMod.user == current_user.uid) or not current_user.can_admin:
        abort(404)
```

Reply:
Good point. I do think it's best if regular users can't even find this view, though, since there's nothing they could do here. So I updated it to this instead (but sanity check my logic here): 

` if not (SubMod.select().where(SubMod.user == current_user.uid) or current_user.can_admin):`